### DISCUSSION TITLE
Apptrian Facebook Pixel

### DISCUSSION THREAD

**Discussion by eneskirca** [breezefront/community#98] [3. Integrations]
Hello 
 
 I am writing to inquire about the compatibility of the Apptrian Facebook Pixel extension with your newly released Breeze Theme.

As an e-commerce business owner, the Apptrian Facebook Pixel extension plays a critical role in tracking user behavior and optimizing our advertising campaigns. With the launch of your Breeze Theme, we are eager to upgrade our website to improve its speed and enhance the overall user experience.

However, in order to continue using this essential extension, we need to ensure its compatibility with your theme. It is important for us to support third-party apps, such as the Apptrian Facebook Pixel, in order to remain competitive and maintain our position as a market leader in the e-commerce industry.(we use Pinterest and tiktok integration extension too)

I would greatly appreciate any information or assistance you can provide regarding this matter. Thank you for your attention to this request, and I look forward to hearing back from you soon.

**Comment by vovayatsyuk** [comment]
Sorry for the delay. Here is the patch for the `app/code/Apptrian/FacebookPixel/view/frontend/templates/code.phtml`:

```diff
--- app/code/Apptrian/FacebookPixel/view/frontend/templates/code.phtml
+++ app/code/Apptrian/FacebookPixel/view/frontend/templates/code.phtml
@@ -50,7 +50,7 @@
 // bof Check if FB Pixel is allowed
 if (in_array($action, $pageHandles) && ($isPixelEnabled || $isApiEnabled)): ?>
 
-<script>
+<script data-breeze>
 require([
     'jquery',
     'mage/url'
@@ -68,9 +68,6 @@
             },
             type: "GET",
             dataType : "json"
-        })
-        .done(function( json ) {
-            var result = json.apptrian_facebook_pixel_matching_section.matching_data;
         });
     }
     
@@ -80,7 +77,7 @@
 </script>
 
 <!-- Facebook Pixel Code -->
-<script>
+<script data-breeze>
     <?php if ($isBaseCodeEnabled): ?>
 !function(f,b,e,v,n,t,s){if(f.fbq)return;n=f.fbq=function(){n.callMethod?
 n.callMethod.apply(n,arguments):n.queue.push(arguments)};if(!f._fbq)f._fbq=n;
@@ -120,7 +117,7 @@
                 return true;
             }
             
-            var settingsUrl = settings.url;
+            var settingsUrl = settings ? settings.url : xhr.response.req.url;
             var sectionName = 'apptrian_facebook_pixel_matching_section';
             var urlPath = 'customer/section/load';
             
@@ -129,7 +126,7 @@
                 && settingsUrl.indexOf(sectionName) !== -1
             ) {
                 propagationStop = 1;
-                var response = JSON.parse(xhr.responseText);
+                var response = JSON.parse(xhr.responseText || xhr.response.text);
                 var section;
                 var sectionData;
                 
@@ -211,9 +208,6 @@
                 },
                 type: "GET",
                 dataType : "json",
-            })
-            .done(function( json ) {
-                var result = json;
             });
         }
```
